expression/builtin_math_vec_test.go

```diff
@@ -95,6 +95,7 @@ var vecBuiltinMathCases = map[string][]vecExprBenchCase{
 		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETDecimal}, geners: nil},
 		{retEvalType: types.ETDecimal, childrenTypes: []types.EvalType{types.ETInt}, geners: nil},
 		{retEvalType: types.ETDecimal, childrenTypes: []types.EvalType{types.ETInt}, childrenFieldTypes: []*types.FieldType{{Tp: mysql.TypeLonglong, Flag: mysql.UnsignedFlag}}, geners: nil},
+		{retEvalType: types.ETDecimal, childrenTypes: []types.EvalType{types.ETDecimal}, geners: nil},
```

Comment:
I have checked the `getFunction` method of the `Floor` expression. You need to make sure the childrenType (Decimal) statifies the conditoin: `fieldTp.Flen-fieldTp.Decimal > mysql.MaxIntWidth-2`. 
So I suggest you to change this test case to:
```
{retEvalType: types.ETDecimal, childrenTypes: []types.EvalType{types.ETDecimal},
  childrenFieldTypes: []*types.FieldType{{Tp: mysql.TypeNewDecimal, Flen: 32, Decimal: 2}}, geners: nil},
```

Reply:
thank you